utils/filesystem/files.go

```diff
@@ -1129,6 +1128,53 @@ func (fs *VFS) LsWithExclusionPatterns(dir string, exclusionPatterns ...string)
 	return
 }
 
+func (fs *VFS) LsRecursive(ctx context.Context, dir string) (files []string, err error) {
+	return fs.LsRecursiveWithExtensionPatternsAndLimits(ctx, dir, NoLimits())
+}
+
+func (fs *VFS) LsRecursiveWithExtensionPatterns(ctx context.Context, dir string, exclusionPatterns ...string) (files []string, err error) {
+	return fs.LsRecursiveWithExtensionPatternsAndLimits(ctx, dir, NoLimits(), exclusionPatterns...)
+}
+
+func (fs *VFS) LsRecursiveWithExtensionPatternsAndLimits(ctx context.Context, dir string, limits ILimits, exclusionPatterns ...string) (files []string, err error) {
+	err = parallelisation.DetermineContextError(ctx)
+	if err != nil {
+		return
+	}
+	if limits == nil {
+		err = fmt.Errorf("%w: missing file system limits", commonerrors.ErrUndefined)
+		return
+	}
+
+	fn := func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		currentDepth, err := FileTreeDepth(fs, dir, path)
+		if err != nil {
+			return err
+		}
+
+		if limits.Apply() && currentDepth >= limits.GetMaxDepth() {
+			return filepath.SkipDir
+		}
+
+		if limits.Apply() && int64(len(files)) >= limits.GetMaxFileCount() {
+			return fmt.Errorf("number of files exceeds the limit of %d: %w", limits.GetMaxFileCount(), commonerrors.ErrTooLarge)
+		}
+
+		if !info.IsDir() {
```

Comment:
this condition check will have to be removed since the description of the interface states `list all files recursively, including subdirectories`

Reply:
I would suggest a parameter flag to be added to the function signature such as `includeDirectories bool`